HIP/hip-1056.md

```diff
@@ -0,0 +1,1769 @@
+
+To enable this, state in the consensus node will be restructured into simple named states. This allows for state changes
+to be expressed as simple CRUD operations on named states. This is so downstream consumers do not have to understand the
+full complexity of the merkle state tree.
+
+## Personas and User Stories
+
+**Personas:**
+- _Downstream Developer_: A provider of value-added services utilizing block streams
+- _Verifier_: Any person or service required to verify the integrity of a block or the information contained within a
+  block.
+
+**User Stories:**
+1. As a downstream developer, I want a streamlined feed that contains all event, transactional, and state details of
+   the network so that I can offer value-added services to the ecosystem.
+2. As a downstream developer, I want each block of data to be verifiable with a single, aggregated signature, so that I
+   can authenticate transactional data without the cost and complexity of downloading multiple signature files for
+   verification.
+3. As a verifier for a network, I want a single, self-contained stream of data with aggregated signature signed by a
+   threshold of stake weights, so that I can cost-effectively confirm that the stream represents the consensus output of
+   the network.
+4. As a downstream developer, I want a low latency feed that provides the state changes associated with transaction(s)
+   within a block, so that I can interact with state changes with the smallest possible delay.
+
+## Specification
+
+The Block Stream will be defined by the continuous transmission of Block information defined as follows.
+- Each `Block` represents all inputs and outputs of the consensus network for zero or more whole `Rounds`. The maximum
+  number of rounds per block will be configurable and start at 1 round per block. It could be increased later if
+  necessary for performance reasons or if round become very frequent and small.
+    - A block *may* be zero rounds because the last block prior to a network "freeze" must communicate the state hash
+      (via a block proof) of the state immediately prior to network shutdown. This is only possible with an extra empty
+      block because the state hash in each block proof is the state hash at the *beginning* of that block/round. This
+      delay of state hash communication is required to simultaneously meet Hedera performance goals and also meet EVM
+      expectations for immediate access to recent (the last 256) block hash values.
+- A `Round` is a sequence of zero or more `Events` that have reached consensus at the same time. Round length in real
+  world time depends on internet latency and event creation rate. In Hedera mainnet today it is around 1 round per
+  second. In the future this could be shorter to lower latency, but would never make sense to be less than ping time.
+- An `Event` is a sequence of zero or more `EventTransactions` submitted by a single node.
```

Comment:
A round is more than just its events, and an event is more than just its transactions. Perhaps it is simplest to say "contains" instead of "is" so we don't have to explain the metadata that comes with each.